Author:         Trent W. Buck <trentbuck@gmail.com>
Forwarded:      http://bugs.darcs.net/patch22
Bug:            http://bugs.darcs.net/issue1554
Bug-Debian:     http://bugs.debian.org/541848
Description:    Resolve issue1554: allow opt-out of -threaded (fix ARM builds).
 GHC doesn't support -threaded on some architectures.
 This patch allows -f-threaded, so Darcs will compile again.
Index: darcs-2.3.0/darcs.cabal
===================================================================
--- darcs-2.3.0.orig/darcs.cabal	2009-11-01 21:19:58.451735608 +1100
+++ darcs-2.3.0/darcs.cabal	2009-11-01 21:20:00.271735867 +1100
@@ -100,6 +100,10 @@
 flag terminfo
   description: Use the terminfo package for enhanced console support.

+flag threaded
+  description: Use threading and SMP support.
+  default:     True
+
 flag type-witnesses
   description: Use GADT type witnesses.
   default:     False
@@ -154,8 +158,10 @@
                     src/Crypt/sha2.c

   -- We need optimizations, regardless of what Hackage says
-  ghc-options:      -Wall -O2 -funbox-strict-fields -threaded
+  ghc-options:      -Wall -O2 -funbox-strict-fields
   ghc-prof-options: -prof -auto-all
+  if flag(threaded)
+    ghc-options:    -threaded

   if !flag(zlib)
     extra-libraries:  z
